src/words/environment/WordsEnvironment.java

```diff
@@ -138,12 +140,26 @@ public WordsObject getObject(String objectName) throws WordsRuntimeException {
 	 * Returns a collection of all objects.
 	 */
 	public Collection<WordsObject> getObjects() {
-		Collection<WordsObject> allObjects = new HashSet<WordsObject>();
-		for (HashMap<String,WordsObject> scopeObjects : gettableObjects) {
-			allObjects.addAll(scopeObjects.values());
+		try {
+			return getObjectsByClass("thing");
+		} catch (WordsClassNotFoundException e) {
+			System.exit(1);
+		}
+		return null;
+	}
+	
+	/**
+	 * Return a collection of all objects of a certain class
```

Comment:
Just for clarity, it might be better to say "of a certain class or any of its subclasses".